.github/actions/test-dotnet/Set-XUnitMaxParallelThreads.ps1

```diff
@@ -1,11 +1,24 @@
 param ($MaxParallelThreads)
 
-Write-Output "Replacing maxParallelThreads in xunit.runner.json files to $MaxParallelThreads."
+Write-Output "Setting maxParallelThreads in xunit.runner.json files to $MaxParallelThreads."
 
-($configFiles = Get-ChildItem "*/**/xunit.runner.json" -Recurse) | ForEach-Object {
-     (Get-Content $_) |
-        ForEach-Object  {$_ -Replace '"maxParallelThreads":\s*([-\d]*)', "`"maxParallelThreads`": $MaxParallelThreads"} |
-        Set-Content $_
+$parameters = @{
+        # -Filter is the fastest way to find files because its value is passed directly to the file system API.
+        Filter = "xunit.runner.json"
+        Recurse = $true
+        # -FollowSymLink is needed because without it errors will be thrown of type: 
+        # "Get-ChildItem: The system cannot find the path specified."
+        FollowSymlink = $true
+        # Ignore warnings that will be printed due to multiple symlinks pointing to the same physical file.
+        WarningAction = "Ignore"
+}
+
+($configFiles = Get-ChildItem @parameters) | ForEach-Object {
+    # Need the parentheses to close the file after reading. Without them, you'll receive an error with Set-Content:
+    # "The process cannot access the file '[...]\xunit.runner.json' because it is being used by another process."
+    (Get-Content $_) |
+        ForEach-Object { $_ -Replace '"maxParallelThreads":\s*([-\d]*)', "`"maxParallelThreads`": $MaxParallelThreads" } |
+            Set-Content $_
```

Comment:
By the way, PS had built-in JSON support for ages.
```pwsh
Get-Content $_ | ConvertFrom-Json
$json.maxParallelThreads = $MaxParallelThreads
ConvertTo-Json -InputObject $json > $_
```

Reply:
This is what the code should have looked like from the beginning!